### PR TITLE
Add tooling for local k3d clusters under tools/dev

### PR DESCRIPTION
**Pull request by trevorwhitney**
**What this PR does / why we need it**:

This PR creates a `tools/dev/k3d` folder, which is a tanka directory structure designed for easily spinning up loki clusters on k3d for development. There is currently one environment `loki-distributed` (based on the helm chart of the same name), but if people are open to this pattern I have more [environments](https://github.com/trevorwhitney/grafana-environments) I'd be happy to PR in. By using `k3d` and `tanka`, we can quickly build local clusters that leverage our open source [jsonnet](https://github.com/grafana/loki/tree/main/production/ksonnet) and/or [helm charts](https://github.com/grafana/helm-charts/), allowing people looking to contribute to Loki to get up and running quickly.

The README has more details, but to use, cd into this folder and:

```bash
make loki-distributed
```

To tear down:

```bash
make down
```

**Checklist**
- [X] Documentation added

### CODE REVIEW

**Review by owen-d — APPROVED** [review]
LGTM after `shellcheck`

File: tools/dev/k3d/lib/grafana/grafana.libsonnet [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    lokiUrl: error 'please provide $._config.lokiUrl',
+    grafana: {
+      datasources: [],
+    },
+  },
+
+  _images+:: {
+    grafana: {
+      repository: 'grafana/grafana',
+      tag: '8.2.5',
```
Comment (MasslessParticle) [inline tools/dev/k3d/lib/grafana/grafana.libsonnet:23]: Grafana's at 8.3.4 these days. Why not latest here?

Comment (trevorwhitney) [reply tools/dev/k3d/lib/grafana/grafana.libsonnet:23]: because copy/paste from somewhere? yes, not a good reason, updating!

**Review by cyriltovena — APPROVED** [review]
LGTM

**Review by chaudum — COMMENTED** [review]
What was the reasoning for using Helm charts via tanka-lib instead of plain jsonnet (or plain Helm)?

**Pull request merged**